Fix jetty-12 tck tests  com.sun.ts.tests.servlet.api.jakarta_servlet_http.httpsessionx.URLClient.invalidateHttpSessionTest and  com.sun.ts.tests.servlet.api.jakarta_servlet_http.httpsessionx.URLClient.invalidateHttpSessionxTest

**Jetty 12**

Test failure:

> Caused by: java.lang.Exception:  Test Test Case Unable to find the following search string in the server's response: 
> 'Test PASSED'
>  at index: 0
>  Server's response:
> -------------------------------------------
> From invalidateHttpSession: Session is not invalidated 
> Test FAILED
> 
> -------------------------------------------

--- a/jetty-ee10/jetty-ee10-servlet/src/main/java/org/eclipse/jetty/ee10/servlet/ServletContextRequest.java
+++ b/jetty-ee10/jetty-ee10-servlet/src/main/java/org/eclipse/jetty/ee10/servlet/ServletContextRequest.java
@@ -696,7 +696,7 @@ public class ServletContextRequest extends ContextRequest
             if (getRequestedSessionId() == null || _coreSession == null)
                 return false;
             //check requestedId (which may have worker suffix) against the actual session id
-            return getSessionManager().getSessionIdManager().getId(getRequestedSessionId()).equals(_coreSession.getId());
+            return _coreSession.isValid() && getSessionManager().getSessionIdManager().getId(getRequestedSessionId()).equals(_coreSession.getId());
         }
 
         @Override

--- a/jetty-ee9/jetty-ee9-nested/src/main/java/org/eclipse/jetty/ee9/nested/Request.java
+++ b/jetty-ee9/jetty-ee9-nested/src/main/java/org/eclipse/jetty/ee9/nested/Request.java
@@ -1511,7 +1511,7 @@ public class Request implements HttpServletRequest
         if (getRequestedSessionId() == null || _coreSession == null)
             return false;
 
-        return (_sessionManager.getSessionIdManager().getId(getRequestedSessionId()).equals(_coreSession.getId()));
+        return (_coreSession.isValid() &&  _sessionManager.getSessionIdManager().getId(getRequestedSessionId()).equals(_coreSession.getId()));
     }
 
     @Override

